rundeckapp/grails-app/domain/rundeck/ScheduledExecution.groovy

```diff
@@ -562,33 +556,65 @@ class ScheduledExecution extends ExecutionContext implements EmbeddedJsonData {
+                                }
+                            }
+                            if(notifMap['plugin']){
+                                def pluginElement=notifMap['plugin']
+                                def plugins=[]
+                                if(pluginElement instanceof Map){
+                                    plugins=[pluginElement]
+                                }else if(pluginElement instanceof Collection){
+                                    plugins= pluginElement
+                                }
+                                plugins.each{ plugin->
+                                    def n=Notification.fromMap(name, plugin)
+                                    if(n){
+                                        nots << n
+                                    }
+                                }
+                            }
                         }
-                    }
-                    if(data.notification[name]['plugin']){
-                        def pluginElement=data.notification[name]['plugin']
-                        def plugins=[]
-                        if(pluginElement instanceof Map){
-                            plugins=[pluginElement]
-                        }else if(pluginElement instanceof Collection){
-                            plugins= pluginElement
-                        }else{
-
+                    }else {
+                        ['urls','email'].each{ subkey->
+                            if(notifications[subkey]){//
+                                Map notificationData = [
+                                        format    : notifications["format"],
+                                        httpMethod: notifications["httpMethod"],
+                                        (subkey)  : notifications[subkey]
+                                ]
+
+                                nots << Notification.fromMap(name, notificationData)
+                            }
                         }
-                        plugins.each{ plugin->
-                            def n=Notification.fromMap(name, plugin)
-                            if(n){
-                                nots << n
+                        if(notifications['plugin']){
+                            def pluginElement=notifications['plugin']
+                            def plugins=[]
+                            if(pluginElement instanceof Map){
+                                plugins=[pluginElement]
+                            }else if(pluginElement instanceof Collection){
+                                plugins= pluginElement
+                            }else{
+
+                            }
+                            plugins.each{ plugin->
+                                def n=Notification.fromMap(name, plugin)
+                                if(n){
+                                    nots << n
+                                }
```

Comment:
This was left to keep compatibility with notifications in yaml definitions of previous versions

Reply:
It can be moved to JobYamlFormat class but it would have to loop two times more over all notifications for each job definition.